staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -122,12 +128,40 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope *RequestSc
 		}
 		options.TypeMeta.SetGroupVersionKind(metav1.SchemeGroupVersion.WithKind("DeleteOptions"))
 
-		span.AddEvent("About to delete object from database")
-		wasDeleted := true
 		userInfo, _ := request.UserFrom(ctx)
 		staticAdmissionAttrs := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Delete, options, dryrun.IsDryRun(options.DryRun), userInfo)
+		deleteValidation := rest.AdmissionToValidateObjectDeleteFunc(admit, staticAdmissionAttrs, scope)
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.AllowUnsafeMalformedObjectDeletion) {
```

Comment:
nearly forgot.  As soon as we decode the delete options above, if the feature is disabled, nil the field.